### ISSUE TITLE
setBreakpoints request path query

### ISSUE THREAD

**Issue by acotty** [eranif/dbgd#8]
As indicated in issue #2 you had issues with the path in the setBreakpoints request. 

I have tried the following combinations with the lldb-vscode.exe and unless I specify just main.cpp the the response is I get "verified": false in the response:
- "D:\\Andrew_Development\\Z_Testing_Apps\\Clang_Multi_File_Test\\main.cpp"
- "/D/Andrew_Development/Z_Testing_Apps/Clang_Multi_File_Test/main.cpp"
- "/Andrew_Development/Z_Testing_Apps/Clang_Multi_File_Test/main.cpp"
- "D:/Andrew_Development/Z_Testing_Apps/Clang_Multi_File_Test/main.cpp"
- "D:\\Andrew_Development\\Z_Testing_Apps\\Clang_Multi_File_Test/main.cpp"

I am going around in circles on this and think I have covered all of the combinations and have not successfully got an absolute path to work.

Can you let me know if you have gotten an absolute path in the setBreakpoints request working? If you have can I please get an example of the path you use and can you also let me know which lldb-vscode.exe you are using and from MSYS2 CLANG64 or LLVM or some other one.

I have not tried the initialize request pathFormat set to 'uri', but if you have not solved the path I will give it a go.

**Comment by eranif** [comment]
for lldb-vscode, i have used the format of /Users/eran/file.cpp

i think that you need to compile your sources with fullpath as well

e.g.
To max lldb, use compilation similar to this:

clang++ C:/Users/eran/file.cpp -fstandalone-debug -gdawrf-2

using -gdwarf-2 with gdb allows you to compile code with gcc and debug it properly with lldb

**Comment by acotty** [comment]
I can get the absolute path to work on Linux and Mac, but still cannot get it to work on Windows.

**Comment by eranif** [comment]
this is how I compile on Windows and it works...

```cmd
C:/msys64/mingw64/bin/mingw32-make.exe -j24 SHELL=cmd.exe -e -f  Makefile
"----------Building project:[ test - Debug ]----------"
mingw32-make[1]: Entering directory 'C:/Users/eran/Downloads/test'
C:/msys64/mingw64/bin/g++.exe  -c  "C:/Users/eran/Downloads/test/main.cpp" -O0 -Wall -gdwarf-2  -o build-Debug/main.cpp.o -I. -I.
C:/msys64/mingw64/bin/g++.exe -o build-Debug\bin\test.exe @build-Debug/ObjectsList.txt -L.
mingw32-make[1]: Leaving directory 'C:/Users/eran/Downloads/test'
=== build completed successfully (0 errors, 0 warnings) ===

```

Note that `-gdwarf-2` is very important. Without it, it won't work

**Comment by eranif** [comment]
When compiled with `-gdwarf-2` (using `g++`) everything work with native path, see attached screenshot:

![bp](https://user-images.githubusercontent.com/3677711/180485911-70ba88a6-f05d-40f6-8798-5f1f5b469a00.png)

**Comment by acotty** [comment]
I now have absolute file path working with g++, but still not working with clang++.

So long as I use g++ I can use lldb-vscode.exe in either of  the mingw64 or the clang64 and it works. 
If I use clang++ then it does not matter if I use either lldb-vscode.exe  it still fails. 

I need to investigate clang++ to see how to produce dwarf-2  debug info in order to see if it can work.

**Comment by eranif** [comment]
when compiling with clang++, you don't need to specify the `-gdwarf-2` since clang++ generates debugging information that is understandable by lldb

However, when using `clang++` to compile code, native paths do not work (at least for me on Windows)
You need to omit the volume part + the drive and use forward slashes:
so:

`c:\path\to\file.cpp` becomes `/path/to/file.cpp`

**Comment by eranif** [comment]
FYI: I released CodeLite 16.2.0 that includes the current implementation for the debugger.
You can use it to compare the protocol exchange between `CodeLite` <-> `lldb-vscode`
(CodeLite generates a specific log file for this under: `%appdata%\CodeLite\debug-adapter-client.log`)

**Comment by acotty** [comment]
Unfortunately for me I have my code on D: and not C:.

Thanks for the codelite info. I have been able to use C::B to debug C::B this afternoon on Windows and am just building the code on the Mac to test it.

**Comment by eranif** [comment]
By default if no drive is specified, then `C:\` is assumed. But since lldb-vscode does not expect the drive at all... I can see how this can be a problem
Seems like a bug in lldb-vscode

**Comment by acotty** [comment]
I tried the official LLVM win64 today and the  lldb-vscode.exe worked different to the MSYS2 version and I stopped using it.

The only consistency with the different DAP debugger exe's is that they work differently and use the MS DAP spec.

I like the codelldb.exe ability to show the comms. I wish the lldb-vscode had the ability to show the comms. The following is the batch file I used to enable debugging with codelldb.exe:

```
set RUST_LOG=debug,codelldb=debug
set RUST_LOG_STYLE=always
set RUST_BACKTRACE=full
E:\Downloads\GCC_Libraries\vscode-lldb\extension\adapter\codelldb.exe --port 12345
```

I download file from the internet to my E:
